### PR TITLE
chore: replacing isort with the isort rule enablement in ruff

### PR DESCRIPTION
**Pull request by abellotti**
chore: replacing isort with the isort rule enablement in ruff

- Removing the lint-isort directive
- Adding the "I" isort rule in ruff
- Fixing one inconsistency between the two
- Removing the isort package dependency from poetry

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -4,9 +4,8 @@
 import sys
 from pathlib import Path
 
-from setuptools import find_packages, setup
-
 from qpc.release import AUTHOR, AUTHOR_EMAIL, ENTRYPOINT, PKG_NAME, URL, VERSION
+from setuptools import find_packages, setup
```
Comment (bruno-fs) [inline setup.py:8]: huh, this is weird. qpc should be in the last section of imports 🤔

Comment (bruno-fs) [reply setup.py:8]: ah, we need a ruff section for isort, configuring qpc as known first party!

Comment (bruno-fs) [reply setup.py:8]: https://beta.ruff.rs/docs/settings/#known-first-party

Comment (abellotti) [reply setup.py:8]: Ahh, thank you.  I'll give that a try.

**Review by infinitewarp — APPROVED** [review]

**Review by bruno-fs — APPROVED** [review]

**Pull request merged**